### PR TITLE
Product dao

### PR DESCRIPTION
**Pull request by Reiterpallasch**
This pull request is strictly to get matt the necessary updates to the product model, albeit the favorites section is finished as well. Products services and DAO are still IP.

### CODE REVIEW

File: server/src/main/java/p2/service/impl/ProductService.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package p2.service.impl;
+import java.util.List;
+import p2.dao.impl.ProductDAO;
+import p2.model.Product;
+
+
+public class ProductService{
```
Comment (mjrerle) [inline server/src/main/java/p2/service/impl/ProductService.java:7]: It is imo cleaner to have a single instance variable called productDAO than have to instantiate a new object in every method

File: server/src/main/java/p2/webservice/FavoriteWebService.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+//			response.getWriter().append(json).close();
+//		} catch (IOException e) {
+//			e.printStackTrace();
+//		}
+		
+	}
+
+
+	public static void update(HttpServletRequest request, HttpServletResponse response) {
+		int FID = Integer.parseInt(request.getParameter("favoriteid"));
```
Comment (mjrerle) [inline server/src/main/java/p2/webservice/FavoriteWebService.java]: these will throw InputMismatchException on request.getParameter returning null (which might happen if the api is called while testing)
also, we can use the new User and Favorite model constructors to save a few api calls here

File: server/src/main/java/p2/webservice/FavoriteWebService.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import p2.model.User;
+import p2.service.impl.FavoriteService;
+import p2.service.impl.ProductService;
+import p2.service.impl.UserService;
+
+public class FavoriteWebService {
+	
+
+	public static void insert(HttpServletRequest request, HttpServletResponse response) {
+		int UID = Integer.parseInt(request.getParameter("customerid"));
```
Comment (mjrerle) [inline server/src/main/java/p2/webservice/FavoriteWebService.java]: use snake case here for consistency (for the parameter name)

File: server/src/main/java/p2/webservice/FavoriteWebService.java (hunk 2 above)
Comment (mjrerle) [reply server/src/main/java/p2/webservice/FavoriteWebService.java]: I dont think we actually need the user and favorite records here, just the ids. Hibernate will just be grabbing the ids from user and favorite anyway to insert or update a row in the table

File: server/src/main/java/p2/service/impl/ProductService.java (hunk 1 above)
Comment (vajirakarunathilake) [reply server/src/main/java/p2/service/impl/ProductService.java]: Also no need to override GenericDAO methods in child classes.

File: server/src/main/java/p2/dao/impl/ProductDAO.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import p2.model.Product;
+
+import java.util.List;
+
+import p2.dao.IProductDAO;
+
+public class ProductDAO extends GenericDAO<Product> implements IProductDAO {
+
+	@Override
+	public int insert(Product t) {
```
Comment (vajirakarunathilake) [inline server/src/main/java/p2/dao/impl/ProductDAO.java:12]: no need to override GenericDAO methods in child classes.

**Review by mjrerle — APPROVED** [review]

**Pull request merged**